src/internal_rep.rs

```diff
@@ -1100,6 +1100,26 @@ impl<'a> Context<'a> {
             mls_high: rename_cow(&self.mls_high, renames),
         }
     }
+
+    // Resolve symbols and return a new Context
+    pub fn resolve(&'_ self, types: &TypeMap, block_context: &BlockContext) -> Self {
+        Context {
+            setype: block_context
+                .symbol_in_context(self.setype.as_ref(), types)
+                .map(|t| Cow::Owned(t.name.to_string()))
+                .or_else(|| {
+                    if self.setype == "this" {
+                        block_context
+                            .get_parent_type_name()
+                            .map(|cs| Cow::Owned(cs.to_string()))
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or_else(|| self.setype.clone()),
+            ..self.clone()
```

Comment:
What is the double '.' here?

Reply:
It's the "struct update syntax".  https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax

Basically, it just means "clone the equivalent field from `self` for the remaining fields`